Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -1068,6 +1071,116 @@ TEST_CASE_METHOD(ReplicatorCollectionSGTest, "Update Once-Conflicted Doc - SGCol
+    const string idPrefix = timePrefix();
+    const string docIDstr = idPrefix + "apera-doc1";
+    const string channelIDa = idPrefix + "a";
+    const string channelIDb = idPrefix + "b";
+
+    constexpr size_t collectionCount = 3;
+    const std::array<C4CollectionSpec, collectionCount> collectionSpecs = {
+            Roses,
+            Tulips,
+            Lavenders
+    };
+
+    // Create a temporary user for this test
+    SG::TestUser testUser { _sg, "aperasg", { channelIDa, channelIDb }, collectionSpecs };
+    _sg.authHeader = testUser.authHeader();
+
+    // Setup pull filter:
+    _pullFilter = [](C4CollectionSpec collectionSpec, C4String docID, C4String revID,
+                     C4RevisionFlags flags, FLDict flbody, void *context) {
+        if ((flags & kRevPurged) == kRevPurged) {
+            ((ReplicatorAPITest*)context)->_counter++;
+            Dict body(flbody);
+            CHECK(body.count() == 0);
+        }
+        return true;
+    };
+
+    // One-shot pull setup
+    std::array<C4Collection*, collectionCount> collections =
+            collectionPreamble(collectionSpecs, testUser);
+    std::vector<C4ReplicationCollection> replCollections {collectionCount};
+
+    for(int i = 0; i < collectionCount; ++i) {
+        replCollections[i] = {
+                collectionSpecs[i], kC4Disabled, kC4OneShot,
+                nullslice, nullptr, _pullFilter, this
+        };
+    }
+
+    ReplParams replParams { replCollections };
+
+    // Setup onDocsEnded:
+    _enableDocProgressNotifications = true;
+    _onDocsEnded = [](C4Replicator* repl,
+                      bool pushing,
+                      size_t numDocs,
+                      const C4DocumentEnded* docs[],
+                      void* context) {
+        for (size_t i = 0; i < numDocs; ++i) {
+            auto doc = docs[i];
+            if ((doc->flags & kRevPurged) == kRevPurged) {
+                ((ReplicatorAPITest*)context)->_docsEnded++;
+            }
+        }
+    };
+
+    // Put doc in remote DB, in channels a and b
+    for(auto& spec : collectionSpecs) {
+        REQUIRE(_sg.upsertDoc( spec, docIDstr, "{}", { channelIDa, channelIDb } ));
+    }
+
+    // Pull doc into CBL:
+    C4Log("-------- Pulling");
+    replicate(replParams);
+
+    // Revoke access to channel 'a':
+    REQUIRE(testUser.setChannels({ channelIDb }));
+
+
+    for(int i = 0; i < collectionCount; ++i) {
+        // Verify
+        c4::ref<C4Document> doc1 = c4coll_getDoc(collections[i], slice(docIDstr), true, kDocGetAll, nullptr);
+        REQUIRE(doc1);
+        CHECK(slice(doc1->revID).hasPrefix("1-"_sl));
+
+        // Update doc to only channel 'b'
+        auto oRevID = slice(doc1->revID).asString();
+        REQUIRE(_sg.upsertDoc(collectionSpecs[i], docIDstr, oRevID, "{}", { channelIDb }));
+    }
+    CHECK(_docsEnded == 0);
+    CHECK(_counter == 0);
```

Comment:
Odd to check them here, as if having anything to do with the preceding for-loop. Suggesting to move up right after replicate.